app/assets/javascripts/controllers/entrypoint_selection_controller.js

```diff
@@ -0,0 +1,104 @@
+/* global miqJqueryRequest */
+
+(function() {
+  var CONTROLLER_NAME = 'entrypointSelectionController';
+
+  var EntrypointSelectionController = function($http, $uibModal) {
+    var vm = this;
+    vm.$http = $http;
+    vm.$uibModal = $uibModal;
+    vm.cache = {};
+
+    vm.$http.get('/tree/automate_entrypoint').then(function(response) {
+      vm.data = response.data;
+    });
+
+    vm.lazyLoad = function(node) {
+      return new Promise(function(resolve) {
```

Comment:
this could be

```
return vm.$http.get('/tree/automate_entrypoint?id=' + encodeURIComponent(node.key))
  .then(function(response) {
    vm.cache[node.key] = response.data;
    return response.data;
  });
```